File: csharp/etl/ETL.cs
using System.Collections.Generic;

namespace Exercism
{
    public class ETL
    {
        /// <summary>
        /// static Transform
        /// Exercism attempt for ETL
        /// I've been reading more about Linq and I plan on refactoring 
        /// this to use System.Linq 
        /// </summary>
        /// <param name="old">a dictionary representing the old point structure</param>
        /// <returns>a dictionary containing the new point structure</returns>
        public static Dictionary<string, int> Transform(Dictionary<int, IList<string>> old)
        {
            var letterPoints = new Dictionary<string, int>();
            
            foreach(var pair in old)
            {
                int point = pair.Key;
                foreach(string letter in pair.Value)
                {
                    letterPoints.Add(letter.ToLower(), point);
                }
            }
             
            return letterPoints;
        }
    }
}

File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {
    
    /**
     * a transform method to convert an old scrabble scoring system to a new one
     * @param old A Map object representing the old scoring system
     * @return A Map object representing the new scoring system
     */
    public Map<String, Integer> transform(Map<Integer, List<String>> old) {
        
        Map<String, Integer> letterPoints = new HashMap<>();
        old.forEach((points, letters) -> 
            letters.forEach(letter -> letterPoints.put(letter.toLowerCase(), points))); 
        
        return letterPoints;
    }
    
}
